packages/react-components/src/components/knowledge-graph/StateManager.tsx

```diff
@@ -1,12 +1,10 @@
 import React, { ReactNode, createContext, useContext, useState } from 'react';
 import { ExecuteQueryCommandOutput } from '@aws-sdk/client-iottwinmaker';
 interface KnowledgeGraphContext {
-  selectedGraphNodeEntityId?: string;
-  setSelectedGraphNodeEntityId: (entityId?: string) => void;
-  queryStatement?: string;
-  setQueryStatement: (query: string) => void;
-  queryResult?: ExecuteQueryCommandOutput;
-  setQueryResult: (result: ExecuteQueryCommandOutput) => void;
+  selectedGraphNodeEntityId?: string | null;
+  setSelectedGraphNodeEntityId: (entityId?: string | null) => void;
+  queryResult?: ExecuteQueryCommandOutput | null;
+  setQueryResult: (result: ExecuteQueryCommandOutput | null) => void;
```

Comment:
I thought about this since you mentioned it last time (This is not part of theis PR) but I have to disagree and here is my reasoning.I think all state managers/stores including React context should give us a central location for our data and the functions which act on our data and thats it they should be agnostic the source of this data (API calls, files ..etc), this makes everything layered, separated and clean. And also the act of querying is asynchronous by nature and asynchronous actions should not be in state manager since state are objects and the action of setting these objects is synchronous process.  I think thats why in Redux the act of querying had to be provided through middleware (like Redux Thunk, etc) that allow for such asynchronous actions. I prefer not to change this for now.

Reply:
I will give it a second thought, try it out and test it in my next PR.